common/scala/src/main/scala/whisk/core/containerpool/logging/SplunkLogStore.scala

```diff
@@ -0,0 +1,158 @@
+  val maxPendingRequests = 500
+
+  val defaultHttpFlow = Http().cachedHostConnectionPoolHttps[Promise[HttpResponse]](
+    host = splunkHost,
+    port = splunkPort,
+    connectionContext =
+      if (disableSNI)
+        Http().createClientHttpsContext(AkkaSSLConfig().mapSettings(s => s.withLoose(s.loose.withDisableSNI(true))))
+      else Http().defaultClientHttpsContext)
+
+  override def fetchLogs(activation: WhiskActivation): Future[ActivationLogs] = {
+
+    //example curl request:
+    //    curl -u  username:password -k https://splunkhost:port/services/search/jobs -d exec_mode=oneshot -d output_mode=json -d "search=search index=\"someindex\" | spath=activation_id | search activation_id=a930e5ae4ad4455c8f2505d665aad282 |  table log_message" -d "earliest_time=2017-08-29T12:00:00" -d "latest_time=2017-10-29T12:00:00"
+    //example response:
+    //    {"preview":false,"init_offset":0,"messages":[],"fields":[{"name":"log_message"}],"results":[{"log_message":"some log message"}], "highlighted":{}}
+    val search =
+      s"""search index="${splunkIndex}"| spath ${activationIdFieldName} | search ${activationIdFieldName}=${activation.activationId.toString} | table ${logMessageFieldName}"""
+
+    val formatter = DateTimeFormatter.ofPattern("YYYY-MM-dd'T'hh:mm:ss").withZone(ZoneId.of("UTC"))
+    val entity = FormData(
+      Map(
+        "exec_mode" -> "oneshot",
+        "search" -> search,
+        "output_mode" -> "json",
+        "earliest_time" -> formatter.format(activation.start),
+        "latest_time" -> formatter.format(activation.end))).toEntity
+
+    log.debug("sending request")
+    queueRequest(
+      Post(splunkApi)
+        .withEntity(entity)
+        .withHeaders(List(Authorization(BasicHttpCredentials(splunkUser, splunkPass))))).flatMap(response => {
+      log.debug(s"splunk API response ${response}")
+      if (response.status.isSuccess()) {
+        Unmarshal(response.entity)
+          .to[String]
+          .map(resultsString => {
+            log.debug(s"splunk API results: ${resultsString}")
+            val jsObject = JsonParser(resultsString).asJsObject
+            //format of results is detailed here: http://docs.splunk.com/Documentation/Splunk/latest/RESTUM/RESTusing#Example_B:_JSON_response_format_example
+            val messages = jsObject
+              .fields("results")
+              .convertTo[JsArray]
+              .elements
+              .map(msgJsValue => {
+                msgJsValue.asJsObject.fields(logMessageFieldName).asInstanceOf[JsString].value
+              })
+            new ActivationLogs(messages)
+          })
+      } else {
+        Future.failed(new RuntimeException(s"failed to read logs from splunk ${response}"))
+      }
+    })
+
+  }
+
+  //based on http://doc.akka.io/docs/akka-http/10.0.6/scala/http/client-side/host-level.html
+  val queue =
+    Source
+      .queue[(HttpRequest, Promise[HttpResponse])](maxPendingRequests, OverflowStrategy.backpressure)
```

Comment:
How are you applying backpressure? `queueRequest` will just continue to run, wouldn't it?

Reply:
I expect Source.queue() to backpressure at queueRequest once the queue buffer is full (hardcoded to 500 currently). SourceQueue.offer() docs indicate failure when backpressure mode is set.

Reply:
I see. Since you're not retrying `offer`, does it make sense to use `dropNew` here (as we do in `CouchDbRestClient`) to explicitly drop the message? I wonder why one would even want the distinction 🤔 .

Reply:
I had expected offer to retry during backpressure mode, so I think you're right. Will change to dropNew.